### ISSUE TITLE
Just being curious about generics

### ISSUE THREAD

**Issue by brennanleez-coder** [nus-cs2030/2021-s2#63]
Hi guys, after having learnt about ArrayList, which is a mutable list,  I have always wondered if there are things that an array can do but an ArrayList cant. My first thought was if it is possible to hard code an ArrayList like a normal array. My attempt is in the picture attached but obviously i failed. Can someone advise whether it is even possible to hard code one or is my syntax.

![Screenshot 2021-03-12 at 11 51 53 PM](https://user-images.githubusercontent.com/80498271/110964247-f640c780-838d-11eb-9a81-05e795f2bc95.png)

**Comment by vishandi** [comment]
hi! I don't think manually code the content of the array will work on ArrayList. This is because ArrayList is a class on its own, so the methods are strict in its class specification, whereas the 'normal' array is built-in to Java.

**Comment by eltonngkk** [comment]
Hi Brennan, ArrayList does not support such kind of declaration in Java. However, you can use the Arrays.asList() method to initialise your ArrayList with values. The code below will work fine. 
ArrayList<String> list = new ArrayList<>(Arrays.asList("abc", "def"))
It is also worthy to note that an Array can store both primitives and objects but an ArrayList can only store objects.

**Comment by Keithlimhs** [comment]
Hi i don't think you can declare an ArrayList in that way, as the API does not support a constructor which constructs an ArrayList in that format. There is a constructor that takes in another Collection object and constructs a new ArrayList with that Collection's elements, i.e ArrayList<Collection<E> c>

**Comment by gableejh** [comment]
I don't think that works for ArrayList, to add elements into the ArrayList you should use .add

**Comment by brennanleez-coder** [comment]
> Hi i don't think you can declare an ArrayList in that way, as the API does not support a constructor which constructs an ArrayList in that format. There is a constructor that takes in another Collection object and constructs a new ArrayList with that Collection's elements, i.e ArrayList<Collection c>

Thank you for your response keith, hope to receive more insights from you whenever I have doubts. Keep living it big.
